sql/src/main/java/io/crate/operation/join/NestedLoopOperation.java

```diff
@@ -78,18 +63,22 @@ public Object get(int index) {
         }
     };
 
-    private final ArrayBlockingQueue<Row> innerRowsQ = new ArrayBlockingQueue<>(1);
+    private RowDownstreamHandle downstream;
+    private volatile int numUpstreams = 0;
```

Comment:
I'm not sure if volatile is correct here?  `numUpstreams++` isn't atomic afaik? So if registerUpstream is called from two threads this might be a nasty race condition